src/main/java/BinaryTree/BinarySearchTree.java

```diff
@@ -0,0 +1,186 @@
+                moveToLeftNode();
+            }
+        }
+
+        if (current.getKey() == dNode.getKey()) {
+
+            // Deleting a node with no children
+            if (current.getLeftNode() == null && current.getRightNode() == null) {
+                if ((Integer)current.getParentNode().getKey() < (Integer)current.getKey()) {
+                    current.getParentNode().setRightNode(null);
+                } else {
+                    current.getParentNode().setLeftNode(null);
+                }
+
+            }
+            // Deleting a node with one child
+            else if (current.getLeftNode() == null && current.getRightNode() != null) {
+                current.getParentNode().setRightNode(current.getRightNode());
+                current.getRightNode().setParentNode(current.getParentNode());
+
+            } else if (current.getLeftNode() != null && current.getRightNode() == null) {
+                current.getParentNode().setLeftNode(current.getLeftNode());
+                current.getLeftNode().setParentNode(current.getParentNode());
+
+            }
+            // Deleting a node with two children
+            else if (current.getLeftNode() != null && current.getRightNode() != null) {
+                moveToRightNode();
+                BinaryNode<T> successor = find_successor();
+                deleteNode(successor);
+                moveToParentNode();
+                successor.setParentNode(current.getParentNode());
+                successor.setLeftNode(current.getLeftNode());
+                successor.setRightNode(current.getRightNode());
+            }
+        }
+        current = original_position;
+    }
+
+    // Get min node in subtree
+    public BinaryNode<T> find_successor() {
+        BinaryNode<T> original_position = current;
+
+        while (current.getLeftNode() != null) {
+            moveToLeftNode();
+        }
+        BinaryNode<T> min_Node = current;
+        current = original_position;
+        return min_Node;
+    }
+
+    @Override
+    public void printDFS() {
+        BinaryNode<T> original_position = current;
+        if (current == null) return;
+
+        current.print();
+
+        if (current.getLeftNode() != null) {
+            moveToLeftNode();
+            printDFS();
+            moveToParentNode();
+        }
+
+        if (current.getRightNode() != null) {
+            moveToRightNode();
+            printDFS();
+            moveToParentNode();
+        }
+
+        current = original_position;
+    }
+
+    @Override
+    public void printBFS() {
+        Queue<BinaryNode<T>> queue = new LinkedList<BinaryNode<T>>() ;
+        if (current == null) return;
+
+        queue.add(current);
+        while (!queue.isEmpty()) {
+            BinaryNode<T> node = queue.remove();
+            node.print();
+            if (node.getLeftNode() != null) queue.add(node.getLeftNode());
+            if (node.getRightNode() != null) queue.add(node.getRightNode());
+        }
+    }
+
```

Comment:
Can you add printInorder()
https://www.geeksforgeeks.org/tree-traversals-inorder-preorder-and-postorder/
output must be 1234567...